### DISCUSSION TITLE
How to use regular expressions with sqlite query?

### DISCUSSION THREAD

**Discussion by realme0213** [SeaQL/sea-orm#687] [Q&A]
I didn't find the relevant documentation about how to use regular expressions with sqlite query, is it supported by `Sea-ORM`?

**Comment by billy1624** [comment]
Hey @realme0213, sorry for the delay. You can use [sea_query::Expr::cust](https://docs.rs/sea-query/0.24.4/sea_query/expr/struct.Expr.html#method.cust) method to construct a custom regular expressions.

**Reply by realme0213** [reply]
Hi, @billy1624  thank you for your reply, I use `sea_query::Expr::cust` method to build my sqlite query

```rust
blog::Entity::find()
    .filter(Expr::col(blog::Column::UserId).eq(user_id))
    .filter(Expr::cust(r#""content" REGEXP "text""#))
```

also, I log raw sqlite string
```sql
SELECT "blogs"."id", "blogs"."user_id", "blogs"."content", "blogs"."title" FROM "blogs" WHERE "user_id" = 'd7f8d349-c44e-4cfa-aa9a-c085f3b0f110' AND "content" REGEXP "text"
```

then, I have a error message:
```bash
Query Error: error returned from database: no such function: REGEXP
```

I found `sea-orm` relying on [SQLx](https://github.com/launchbadge/sqlx), and `SQLx` doesn't seem to support sqlite REGEXP, https://github.com/launchbadge/sqlx/issues/194

Here is some informations about sqlite REGEXP from [sqlite document](https://www.sqlite.org/search?s=d&q=regexp)
> The REGEXP operator is a special syntax for the regexp() user function. No regexp() user function is defined by default and so use of the REGEXP operator will normally result in an error message. If an [application-defined SQL function](https://www.sqlite.org/appfunc.html) named "regexp" is added at run-time, then the "X REGEXP Y" operator will be implemented as a call to "regexp(Y,X)".

So, is there any other workaround to support regular expressions with `sea-orm` sqlite query
